File: yeon/data_structure/18258_BJ.py
import sys
from collections import deque

N = int(sys.stdin.readline().rstrip())
q = deque()
for _ in range(N):
  input = sys.stdin.readline().rstrip().split()
  if input[0]=="push" :
    q.append(int(input[1]))
  elif input[0]=="pop":
    if not q: 
      print(-1)
      continue
    print(q.popleft())
  elif input[0]== "size" :
    print(len(q))
  elif input[0]== "empty" :
    if not q: print(1)
    else: print(0)
  elif input[0]== "front" :
    if not q:
      print(-1)
      continue
    print(q[0])
  elif input[0] == "back" :
    if not q:
      print(-1)
      continue
    print(q[-1])

    
    
    
    

File: Elmo/data_structure/18258_BJ.cpp
#include <iostream>
using namespace std;
int arr[2000001];
int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    cout.tie(NULL);
    int n, l=0, front = 0; cin >> n;
    for (int i = 0; i < n; i++) {
        string s; cin >> s;
        if (s == "push") {
            int x; cin >> x;
            arr[l++] = x;
        }
        else if (s == "front") {
            if (l > front) cout << arr[front] << "\n";
            else cout << -1 << "\n" ;
        }
        else if (s == "back") {
            if (l > front) cout << arr[l-1] << "\n";
            else cout << -1 << "\n";
        }
        else if (s == "size") {
            cout << l-front << "\n";
        }
        else if (s == "empty") {
            if (l == front) cout << 1 << "\n";
            else cout << 0 << "\n";
        }
        else if (s == "pop") {
            if (l > front) cout << arr[front++] << "\n";
            else cout << -1 << "\n";
        }
    }

}